Forge/src/main/java/me/ferdz/placeableitems/client/model/GlassBottleFluidModel.java

```diff
@@ -0,0 +1,39 @@
+package me.ferdz.placeableitems.client.model;
+
+import me.ferdz.placeableitems.block.PlaceableItemsBlock;
+import me.ferdz.placeableitems.block.component.impl.BiPositionBlockComponent;
+import me.ferdz.placeableitems.client.AllVertexBoundingBox;
+import me.ferdz.placeableitems.init.PlaceableItemsBlockRegistry;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.client.renderer.Vector3d;
+import net.minecraft.util.Direction;
+
+/**
+ * An implementation of {@link FluidModel} for the {@link PlaceableItemsBlockRegistry#GLASS_BOTTLE}.
+ *
+ * @author Parker Hawke - Choco
+ */
+public class GlassBottleFluidModel extends FluidModel {
+
+    private static final Vector3d ROTATION_POINT = new Vector3d(8.0D, 1.0D, 8.0D);
+    private static final Vector3d UP_ROTATION_POINT = new Vector3d(8.0D, 5.0D, 8.0D);
+
+    private static final AllVertexBoundingBox BOUNDS = AllVertexBoundingBox.fromAABB(-2.5 / 16.0, 0.0D, -2.5 / 16.0, 2.5 / 16.0, 4 / 16.0, 2.5 / 16.0);
+
+    @Override
+    public AllVertexBoundingBox calculateRenderBounds(BlockState state) {
+        return BOUNDS.rotateAroundY(Math.toRadians((-state.get(PlaceableItemsBlock.ROTATION) * 360F / 16.0F)));
+    }
+
+    @Override
+    public Vector3d getRotationPoint(BlockState state) {
+        return state.get(BiPositionBlockComponent.UP) ? UP_ROTATION_POINT : ROTATION_POINT;
```

Comment:
Since you're rotating about the Y axis, and that the bottle is centered on both UP and DOWN models, why is this important?

Reply:
Also this should be checked, there are no guarantees that this BlockState will always contain the UP Property

Reply:
The rotation point is used also as an offset. If they were the same, block states with the "UP" property set to true would have their fluids rendered in the same position as ones set to false. It's used to directly modify the arguments of `GlStateManager.translated()` by increasing by the supplied values.

As for it requiring a check, it doesn't. Because the GlassBottleFluidModel is bound to the `GLASS_BOTTLE` PlaceableItemBlock, it's guaranteed to have this property as it's part of the state container. If, however, this model were bound to another block that _DOES_ have this property, then yes a check would be required. That being said, it's expected that most blocks have their own FluidModels as they'd all have unique bounds and rotation points.

Reply:
I see. In that case I think `RotationPoint` might not be fully accurate, since you also use this point for translation. Any idea of a less confusing name? I'm thinking Origin but since it's not truly the origin I'm not sure it works

Reply:
Before rotation point it had been named offset, though that too seems strange to me and is ultimately the reason I had changed it to rotation point. I'm not opposed to origin.

Reply:
I'm thinking maybe rotationOrigin or rotationAxis. A little documentation saying that the Y param is used for offset wouldn't hurt either

Reply:
I honestly preferred just `origin` because `rotationOrigin` implies it exists solely for rotation, which isn't the case. It does also offset the rendering bounds. I've changed it to that instead 😄